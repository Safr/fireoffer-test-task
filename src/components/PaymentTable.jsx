import React from 'react';
import PropTypes from 'prop-types';
import moment from 'moment';
import DateRangeIcon from '@material-ui/icons/DateRange';
import AttachMoneyIcon from '@material-ui/icons/AttachMoney';
import AssignmentIcon from '@material-ui/icons/Assignment';
import AccountBalanceWalletIcon from '@material-ui/icons/AccountBalanceWallet';
import { withStyles } from '@material-ui/core/styles';
import Table from '@material-ui/core/Table';
import TableBody from '@material-ui/core/TableBody';
import TableCell from '@material-ui/core/TableCell';
import TableHead from '@material-ui/core/TableHead';
import TableRow from '@material-ui/core/TableRow';

const CustomTableCell = withStyles(theme => ({
  head: {
    backgroundColor: '#242424',
    color: theme.palette.common.white,
  },
  body: {
    fontSize: 14,
  },
}))(TableCell);


const styles = theme => ({
  table: {
    width: '100%',
    borderCollapse: 'collapse',
  },
  th: {
    position: 'sticky',
    top: 0,
    fontSize: '16px',
  },
  td: {
    borderLeft: '1px solid rgba(224, 224, 224, 1)',
  },
  iconWrapper: {
    display: 'flex',
    alignItems: 'center',
  },
  icon: {
    marginLeft: theme.spacing.unit * 1.2,
    fontSize: 18,
  },
});

const renderPaymentInfo = (payments, td) => {
  let id;
  let date;
  let sum;
  let purpose;
  let balance;
  return payments.map((paymentItem) => {
    id = paymentItem.id
    date = paymentItem.date;
    sum = paymentItem.sum;
    purpose = paymentItem.purpose;
    balance = paymentItem.balance;
    return (
      <TableRow key={id}>
        <TableCell component="td" scope="row">
        {moment(date).format('YYYY-MM-DD')}
        </TableCell>
        <TableCell className={td}component="td" scope="row">{sum}</TableCell>
        <TableCell className={td}component="td" scope="row">{purpose}</TableCell>
        <TableCell className={td}component="td" scope="row">{balance}</TableCell>
      </TableRow>
    );
  });
};

const PaymentTable = ({ payments, classes }) => {
  return (
    <div className="table-wrapper">
      <Table className={classes.table}>
        <TableHead>
          <TableRow>
            <CustomTableCell className={classes.th}>
            <span className={classes.iconWrapper}>
              Date
              <DateRangeIcon className={classes.icon} />
              </span>
            </CustomTableCell>
            <CustomTableCell className={classes.th}>
            <span className={classes.iconWrapper}>
              Sum
              <AttachMoneyIcon className={classes.icon} />
              </span>
            </CustomTableCell>
            <CustomTableCell className={classes.th}>
            <span className={classes.iconWrapper}>
              Purpose
              <AssignmentIcon className={classes.icon} />
              </span>
            </CustomTableCell>
             <CustomTableCell className={classes.th}>
            <span className={classes.iconWrapper}>
              Balance
              <AccountBalanceWalletIcon className={classes.icon}/>
            </span>
            </CustomTableCell>
          </TableRow>
        </TableHead>
       <TableBody>
          {renderPaymentInfo(payments, classes.td, classes.icon)}
        </TableBody>
      </Table>
    </div>
  );
};

PaymentTable.propTypes = {
  payments: PropTypes.arrayOf(PropTypes.shape({
    id: PropTypes.string.isRequired,
    date: PropTypes.string.isRequired,
    sum: PropTypes.string.isRequired,
    purpose: PropTypes.string.isRequired,
    balance: PropTypes.string.isRequired,
  })),
  classes: PropTypes.shape({
    table: PropTypes.string.isRequired,
    td: PropTypes.string.isRequired,
    th: PropTypes.string.isRequired,
  }),
};

export default withStyles(styles)(PaymentTable);
