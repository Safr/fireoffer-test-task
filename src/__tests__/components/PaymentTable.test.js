import React from 'react';
import { shallow } from 'enzyme';
import PaymentTable from '../../components/PaymentTable';
import payments from '../../__mocks__/apiResponse';

describe('PaymentTable', () => {
  test('should render PaymentTable correctly without props', () => {
    const wrapper = shallow(<PaymentTable payments={[]} classes={{}} />);
    expect(wrapper).toMatchSnapshot();
  });
  test('should render PaymentTable correctly with props', () => {
    const wrapper = shallow(<PaymentTable payments={payments} classes={{}} />);
    expect(wrapper).toMatchSnapshot();
  });
    
});
