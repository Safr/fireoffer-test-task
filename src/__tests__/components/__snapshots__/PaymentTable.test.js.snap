// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PaymentTable should render PaymentTable correctly with props 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <WithStyles(PaymentTable)
    classes={Object {}}
    payments={
      Array [
        Object {
          "balance": "$3,830.90",
          "date": "2018-09-28",
          "id": "5bcaf299081302b4b3edf7e0",
          "purpose": "bill",
          "sum": "$558.00",
        },
        Object {
          "balance": "$2,408.67",
          "date": "2017-11-30",
          "id": "5bcaf2995e06351ef89f4647",
          "purpose": "bill",
          "sum": "$866.00",
        },
      ]
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "classes": Object {
        "icon": "PaymentTable-icon-5",
        "iconWrapper": "PaymentTable-iconWrapper-4",
        "table": "PaymentTable-table-1",
        "td": "PaymentTable-td-3",
        "th": "PaymentTable-th-2",
      },
      "payments": Array [
        Object {
          "balance": "$3,830.90",
          "date": "2018-09-28",
          "id": "5bcaf299081302b4b3edf7e0",
          "purpose": "bill",
          "sum": "$558.00",
        },
        Object {
          "balance": "$2,408.67",
          "date": "2017-11-30",
          "id": "5bcaf2995e06351ef89f4647",
          "purpose": "bill",
          "sum": "$866.00",
        },
      ],
    },
    "ref": null,
    "rendered": null,
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "classes": Object {
          "icon": "PaymentTable-icon-5",
          "iconWrapper": "PaymentTable-iconWrapper-4",
          "table": "PaymentTable-table-1",
          "td": "PaymentTable-td-3",
          "th": "PaymentTable-th-2",
        },
        "payments": Array [
          Object {
            "balance": "$3,830.90",
            "date": "2018-09-28",
            "id": "5bcaf299081302b4b3edf7e0",
            "purpose": "bill",
            "sum": "$558.00",
          },
          Object {
            "balance": "$2,408.67",
            "date": "2017-11-30",
            "id": "5bcaf2995e06351ef89f4647",
            "purpose": "bill",
            "sum": "$866.00",
          },
        ],
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;

exports[`PaymentTable should render PaymentTable correctly without props 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <WithStyles(PaymentTable)
    classes={Object {}}
    payments={Array []}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "classes": Object {
        "icon": "PaymentTable-icon-5",
        "iconWrapper": "PaymentTable-iconWrapper-4",
        "table": "PaymentTable-table-1",
        "td": "PaymentTable-td-3",
        "th": "PaymentTable-th-2",
      },
      "payments": Array [],
    },
    "ref": null,
    "rendered": null,
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "classes": Object {
          "icon": "PaymentTable-icon-5",
          "iconWrapper": "PaymentTable-iconWrapper-4",
          "table": "PaymentTable-table-1",
          "td": "PaymentTable-td-3",
          "th": "PaymentTable-th-2",
        },
        "payments": Array [],
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
